unittests/snapshot_tests.cpp

```diff
@@ -250,91 +362,83 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(test_replay_over_snapshot, SNAPSHOT_SUITE, snapsho
+   const auto replay_head = replay_chain.control->head_block_num();
+   auto snap_head = snap_chain.control->head_block_num();
+   BOOST_REQUIRE_EQUAL(replay_head, snap_chain.control->last_irreversible_block_num());
+   for (auto block_num = replay_head + 1; block_num <= snap_head; ++block_num) {
+      auto block = snap_chain.control->fetch_block_by_number(block_num);
+      replay_chain.push_block(block);
+   }
+   verify_integrity_hash<SNAPSHOT_SUITE>(*chain.control, *replay_chain.control);
 
-         // diff common
-         auto common_count = std::min(l_arr.size(), r_arr.size());
-         for (size_t idx = 0; idx < common_count; idx++) {
-            const auto& l_val = l_arr.at(idx);
-            const auto& r_val = r_arr.at(idx);
-            variant_diff_helper(l_val, r_val, [&](const std::string& path, const fc::variant& lhs, const fc::variant& rhs){
-               out( std::string("[") + std::to_string(idx) + std::string("]") + path, lhs, rhs);
-            });
-         }
+   auto block = chain.produce_block();
+   chain.control->abort_block();
+   snap_chain.push_block(block);
+   replay_chain.push_block(block);
+   verify_integrity_hash<SNAPSHOT_SUITE>(*chain.control, *snap_chain.control);
+   verify_integrity_hash<SNAPSHOT_SUITE>(*chain.control, *replay_chain.control);
+
+   snapshotted_tester replay2_chain(snap_chain.get_config(), SNAPSHOT_SUITE::get_reader(snapshot), ordinal++, config_file_handling::copy_config_files);
+   const auto replay2_head = replay2_chain.control->head_block_num();
+   snap_head = snap_chain.control->head_block_num();
+   BOOST_REQUIRE_EQUAL(replay2_head, snap_chain.control->last_irreversible_block_num());
+   for (auto block_num = replay2_head + 1; block_num <= snap_head; ++block_num) {
+      auto block = snap_chain.control->fetch_block_by_number(block_num);
+      replay2_chain.push_block(block);
+   }
+   verify_integrity_hash<SNAPSHOT_SUITE>(*chain.control, *replay2_chain.control);
+
+   // verifies that chain's block_log has a genesis_state (and blocks starting at 1)
+   controller::config copied_config = copy_config_and_files(chain.get_config(), ordinal++);
+   auto genesis = chain::block_log::extract_genesis_state(chain.get_config().blocks_dir);
+   BOOST_REQUIRE(genesis);
+   tester from_block_log_chain(copied_config, *genesis);
+   const auto from_block_log_head = from_block_log_chain.control->head_block_num();
+   BOOST_REQUIRE_EQUAL(from_block_log_head, snap_chain.control->last_irreversible_block_num());
+   for (auto block_num = from_block_log_head + 1; block_num <= snap_head; ++block_num) {
+      auto block = snap_chain.control->fetch_block_by_number(block_num);
+      from_block_log_chain.push_block(block);
+   }
+   verify_integrity_hash<SNAPSHOT_SUITE>(*chain.control, *from_block_log_chain.control);
+
+   // verifies that snap_chain's block_log does not have a genesis_state
```

Comment:
This part of the test isn't really checking that the snap_chain's block log does not have a genesis state. The failure it is encountering is do to the instantiation of tester calling the overload of `controller_impl::startup` that expects an existing state database even though one does not exist because the `copied_config` points to a fresh new `state` directory.

Reply:
Not sure how to (or if there is a point to) test this, since I think that the most interesting part of this would be in chain_plugin, I'm just pulling this out.

Reply:
I'm okay with that.